src/hotspot/share/utilities/macros.hpp

```diff
@@ -28,6 +28,10 @@
 // Use this to mark code that needs to be cleaned up (for development only)
 #define NEEDS_CLEANUP
 
+// For when Pragmas need to be reliably nested in macros. Also handles
+// quotation marks, so use sites are not required to pass string literals
+#define PRAGMA(str) _Pragma(#str)
```

Comment:
This shouldn't be here.  A file could include only compilerWarnings.hpp and not this file, and not be able to use pragma macros provided by compilerWarnings.hpp.

Reply:
@kimbarrett compilerWarnings.hpp does include macros.hpp before the compiler specific macros, so this shouldn't be an issue